com.unity.ml-agents.extensions/Runtime/Input/InputActionActuator.cs

```diff
@@ -0,0 +1,108 @@
+#if MLA_INPUT_SYSTEM
+
+#define ACTUATOR_DEBUG
+#undef ACTUATOR_DEBUG
+using Unity.MLAgents.Actuators;
+using UnityEngine;
+using UnityEngine.InputSystem;
+
+
+#if ACTUATOR_DEBUG
+using UnityEngine.EventSystems;
+using UnityEngine.InputSystem.Controls;
+using UnityEngine.InputSystem.LowLevel;
+#endif
+
+namespace Unity.MLAgents.Extensions.Runtime.Input
+{
+    public class InputActionActuator : IActuator, IHeuristicProvider
+    {
+        bool m_IsInHeuristicMode;
+        InputAction m_Action;
+        IRLActionInputAdaptor m_InputAdaptor;
+        InputDevice m_Device;
+        InputControl m_Control;
+        Agent m_Agent;
+
+#if ACTUATOR_DEBUG
+        float m_Time;
+        bool m_Flip;
+#endif
+
+        public InputActionActuator(Agent agent,
+            InputAction action,
+            IRLActionInputAdaptor adaptor,
+            bool isInHeuristicMode)
+        {
+            m_Agent = agent;
+            m_IsInHeuristicMode = isInHeuristicMode;
```

Comment:
Can this change on the fly for a given Agent?

Reply:
I can poll agent instead.